app/services/doc_auth/socure/requests/document_request.rb

```diff
@@ -47,6 +47,43 @@ def handle_http_response(http_response)
           JSON.parse(http_response.body, symbolize_names: true)
         end
 
+        def handle_invalid_response(http_response)
+          message = [
+            self.class.name,
+            'Unexpected HTTP response',
+            http_response.status,
+          ].join(' ')
+          exception = DocAuth::RequestError.new(message, http_response.status)
+
+          response_body = begin
+            http_response.body.present? ? JSON.parse(http_response.body) : {}
+          rescue JSON::JSONError
+            {}
+          end
+
+          handle_connection_error(
+            exception: exception,
+            status_code: response_body.dig('status', 'code'),
+            status_message: response_body.dig('status', 'message'),
+          )
```

Comment:
I tested receiving a 401 from Socure by providing a bad token in the request, which resulted in a 500.  Please address this.
<img width="1488" alt="Screenshot 2024-11-01 at 12 04 36 PM" src="https://github.com/user-attachments/assets/f512297a-36b2-400b-984b-9d0c6e511023">

Reply:
@amirbey should be resolved as a part of [278e169]